### DISCUSSION TITLE
ARD Noise variance

### DISCUSSION THREAD

**Discussion by patel-zeel** [SheffieldML/GPy#952] [Q&A]
We often assume constant noise variance in standard GPs or sometimes [heteroscedastic noise](https://icml.cc/Conferences/2011/papers/456_icmlpaper.pdf) that depends on the input vector (with all features). Why we do not have ARD like behaviour for noise variance where noise variance in observations with respect to each feature can be different? I'd greatly appreciate an intuitive/mathematical explaination.

**Comment by ekalosak** [comment]
To be fair, ARD introduces a per-dimension lenghtscale parameter that _does_ change the covariance-per-unit of each feature. "ARD like behavior" as such may be a red herring.

That being said, heteroskedastic GPs are supported today, see e.g. this notebook: https://nbviewer.org/github/SheffieldML/notebook/blob/master/GPy/heteroscedastic_regression.ipynb
